commandapi-core/src/main/java/dev/jorel/commandapi/arguments/AbstractArgument.java

```diff
@@ -276,6 +285,42 @@ public Impl setOptional(boolean optional) {
 		return instance();
 	}
 
+	/**
+	 * Returns a list of arguments linked to this argument.
+	 *
+	 * @return A list of arguments linked to this argument
+	 */
+	public List<Argument> getCombinedArguments() {
+		return combinedArguments;
+	}
+
+	/**
+	 * Returns true if this argument has linked arguments.
+	 *
+	 * @return true if this argument has linked arguments
+	 */
+	public boolean hasCombinedArguments() {
+		return !combinedArguments.isEmpty();
+	}
+
+	/**
+	 * Adds linked arguments to this argument. Linked arguments are used to have required arguments after optional arguments
+	 * by ignoring they exist until they are added to the arguments array for registration
+	 *
+	 * @param combinedArguments The arguments to link
+	 * @return this current argument
+	 */
+	@SafeVarargs
+	public final Impl combineWith(Argument... combinedArguments) {
+		for (Argument argument : combinedArguments) {
+			argument.resetRequirements();
+			argument.withRequirement(this.requirements);
+			argument.withPermission(this.permission);
+			this.combinedArguments.add(argument);
+		}
+		return instance();
+	}
+
```

Comment:
Should the requirements and permission of the parent Argument be set right now? If I did:
```java
new StringArgument("base").combineWith(
	new StringArgument("child")
).withPermission("permission")
```
`StringArgument("child")` wouldn't get the `"permission"` permission. Same if I added a requirement after adding a combined argument. Maybe permissions and requirements could be applied to the child arguments when unpacking the combined arguments?

Reply:
That’s a good point. I didn‘t think about that.

Reply:
On a second thought, I think this is useful. Consider this:
- Permissions and requirements are added to the first argument
  - they will be copied over
- Permissions and requirements are added to the second argument
  - they will be overwritten
In both cases it will result in both arguments having the same permissions and requirements.